src/users/components/search-filter-users/search-filter-users.component.ts

```diff
@@ -0,0 +1,152 @@
+import { Component, EventEmitter, Input, OnDestroy, OnInit, Output } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { Observable, Subject, of } from 'rxjs';
+import { debounceTime, distinctUntilChanged, filter, map, switchMap, takeUntil, tap } from 'rxjs/operators';
+
+interface User {
+  firstName: string;
+  lastName: string;
+  email: string;
+}
+
+@Component({
+  selector: 'app-search-filter-users',
+  templateUrl: './search-filter-users.component.html',
+  styleUrls: ['./search-filter-users.component.scss']
+})
+export class SearchFilterUserComponent implements OnInit, OnDestroy{
+  @Output() filterValues = new EventEmitter<{ name: string; email: string; status: string }>();
+  @Input() usersList: User[] = [];
+
+  filteredJudicialUsers$: Observable<User[]>;
+  searchFilterUserForm: FormGroup;
+  nameFilterControl: FormControl;
```

Comment:
Have you considered using a static typed form here instead? It would be easier to work with the form

Reply:
Added types